core/src/main/java/com/amplifyframework/auth/AuthException.java

```diff
@@ -191,4 +191,384 @@ public UnknownException(Throwable cause) {
+
+            /**
+             * Default message/recovery suggestion with a cause.
+             * @param cause The original error.
+             */
+            public CodeMismatchException(Throwable cause) {
+                super(MESSAGE, cause, RECOVERY_SUGGESTION);
+            }
+        }
+
+        /**
+         * Could not perform the action because confirmation code has expired.
+         */
+        public static class CodeExpiredException extends AWSCognitoAuthException {
+            private static final long serialVersionUID = 1L;
+            private static final String MESSAGE = "Confirmation code has expired.";
+            private static final String RECOVERY_SUGGESTION = "Retry operation and send another confirmation code.";
+
+            /**
+             * Default message/recovery suggestion with a cause.
+             * @param cause The original error.
+             */
+            public CodeExpiredException(Throwable cause) {
+                super(MESSAGE, cause, RECOVERY_SUGGESTION);
+            }
+        }
+
+        /**
+         * Could not perform the action because there exists incorrect parameters.
+         */
+        public class InvalidParameterException extends AWSCognitoAuthException {
+            private static final long serialVersionUID = 1L;
+            private static final String MESSAGE = "One or more parameters are incorrect.";
+            private static final String RECOVERY_SUGGESTION = "Enter correct parameters.";
+
+            /**
+             * Default message/recovery suggestion with a cause.
+             * @param cause The original error.
+             */
+            public InvalidParameterException(Throwable cause) {
+                super(MESSAGE, cause, RECOVERY_SUGGESTION);
+            }
+        }
+
+        /**
+         * Could not perform the action because the password given is invalid.
+         */
+        public static class InvalidPasswordException extends AWSCognitoAuthException {
+            private static final long serialVersionUID = 1L;
+            private static final String MESSAGE = "The password given is invalid.";
+            private static final String RECOVERY_SUGGESTION = "Enter correct password.";
+
+            /**
+             * Default message/recovery suggestion with a cause.
+             * @param cause The original error.
+             */
+            public InvalidPasswordException(Throwable cause) {
+                super(MESSAGE, cause, RECOVERY_SUGGESTION);
+            }
+        }
+
+        /**
+         * Could not perform the action because number of allowed operation has exceeded.
+         */
+        public class LimitExceededException extends AWSCognitoAuthException {
+            private static final long serialVersionUID = 1L;
+            private static final String MESSAGE = "Number of allowed operation has exceeded.";
+            private static final String RECOVERY_SUGGESTION = "Turn off and re-attempt this operation.";
```

Comment:
Hmm - this recovery suggestion seems wrong. I'd expect it'd be something along the lines of increasing their service limit or waiting awhile before re-attempting.